### DISCUSSION TITLE
`hreflang` attribute not lowercased after build

### DISCUSSION THREAD

**Discussion by ihmpavel** [vercel/next.js#39226] [Ideas]
Hi, I came across different behavior of `hrefLang` 

The attribute should be lowercased after ssr render, but is not changed

Example
```
<link
  rel='alternate'
  hrefLang='en'
  href='...'
/>
```

Expected output
```
<link
  rel='alternate'
  hreflang='en'
  href='...'
/>
```

Is is a bug? 

_(I know HTML attributes are case insensitive, but some crawlers are unable to parse HTML with `hreflang` which is not lowercased)_

**Comment by princetanwar** [comment]
Hi @ihmpavel,

I'm also facing the same issue did you find any solution?

when I forcefully make it "hreflang" it shows correctly in the page source but it is removed in the inspect element.

**Comment by eliawk** [comment]
Yep, same problem here!

**Comment by kliatskiidenis** [comment]
Greetings
Has anyone found a solution, is this not a problem, because I, too, am developing on Nextjs 14 with app routing, I do as indicated in the official documentation and I get hrefLang instead of hreflang (in lower case)
<img width="692" alt="Знімок екрана 2024-09-05 о 11 42 23" src="https://github.com/user-attachments/assets/7f3c0182-a905-4286-a473-6f7c1dcf3a4d">


<img width="475" alt="Знімок екрана 2024-09-05 о 11 34 29" src="https://github.com/user-attachments/assets/3a0101d1-189d-43cf-ae0c-fdcfb012e3cc">

**Comment by yandodov** [comment]
I can reproduce this on both Next `14.2.11` (latest 14 release at the time of writing) and Next `15.0.0-canary.156`.

@leerob could you turn this discussion to an issue? [The HTML spec defines a `hreflang` attribute](https://html.spec.whatwg.org/multipage/links.html#attr-hyperlink-hreflang), not a `hrefLang` one.

**Reply by todor0v** [reply]
@hdodov FYI, there is an [existing PR](https://github.com/vercel/next.js/pull/66983) that has not been merged yet.
